apps/designer/app/designer/features/sidebar-left/panels/components/image.tsx

```diff
@@ -18,21 +17,60 @@ const useImageWithFallback = ({ path }: { path: string }) => {
 };
 
 export const AssetManagerImage = ({
-  asset: { path, alt },
+  path,
+  alt,
+  uploading,
 }: {
-  asset: Asset;
+  path: string;
+  alt?: string;
+  uploading?: boolean;
 }) => {
   const src = useImageWithFallback({ path });
+  const [progressBarPercentage, setProgressBarPercentage] = useState(0);
+
+  useEffect(() => {
+    if (uploading) {
+      setInterval(() => {
```

Comment:
missing a cleanup, also react-use has this in a cleaned up version, maybe its better to use it?

Reply:
danger is that user closes, component unmounts, this still still runs and results in warnings from react

Reply:
also done!